crates/rustc_codegen_spirv/src/builder_spirv.rs

```diff
@@ -25,6 +25,13 @@ pub enum SpirvValueKind {
     /// of such constants, instead of where they're generated (and cached).
     IllegalConst(Word),
 
+    /// This can only happen in one specific case - which is as a result of
+    /// `codegen_buffer_store_intrinsic`, that function is supposed to return
+    /// OpTypeVoid, however because it gets inline by the compiler it can't.
+    /// Instead we return this, and trigger an error if we ever end up using the
+    /// result of this function call (which we can't).
+    IllegalTypeUsed(Word),
```

Comment:
I'm not sure I understand why this is needed, wouldn't an `undef` work? I thought that was what was used in other similar cases. It's not that important though, feel free to merge without touching this at all.

Reply:
Ah, yeah, probably, a zombie'd undef would work (I'm guessing you can't have an undef void) - this was directly carried over from the previous implementation, though, didn't really think about it that much.

Reply:
For context, there's some discussion around the original implementation here https://github.com/EmbarkStudios/rust-gpu/pull/450 (although quite a bit of discussion isn't actually in that PR, due to instead having calls due to frustration around PR stuff)

Reply:
Maybe what happens with the other cases is they make an `undef` `()` (i.e. the ZST Rust type, an `OpTypeStruct` without fields), and `OpTypeVoid` appears here only because it was intentionally specified.

~~This gives me an idea I could try (removing `SpirvType::Void` and using `Option` where necessary instead)~~ - tho because `rspirv` doesn't bother interning types, it would be unnecessarily inefficient.